src/MessageBroker.php

```diff
@@ -170,17 +173,28 @@ public function publishMessage($payload, $deliveryMode = 1) {
       $channel->queue_bind($queueOption['name'], $this->exchangeOptions['name'], $queueOption['bindingKey']);
     }
 
-    // @todo: 'delivery_mode' needs to be a setting as not all messages will require acknowledgement
+    // Routing key value can be a parameter to publishMessage() or a setting in the settings array sent
+    // to the instantiation of the Message Broker object.
+    $routingKey = isset($routingKey) ? $routingKey : $this->routingKey;
     $messageProperties = array(
       'delivery_mode' => $deliveryMode,
     );
     $message = new AMQPMessage($payload, $messageProperties);
-    $channel->basic_publish($message, $this->exchangeOptions['name'], $this->routingKey);
+    $channel->basic_publish($message, $this->exchangeOptions['name'], $routingKey);
 
     $channel->close();
   }
 
   /**
+   * publishMessage - DEPRECIATED
+   *
+   * Replaced by publish()
+   */
+  public function publishMessage($payload, $deliveryMode = 1, $routingKey) {
```

Comment:
:+1:! I'd also recommend using the [`@deprecated` PHPDoc tag](http://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.deprecated.pkg.html), since some editors (like PHPStorm) will pick up on that and warn when using a deprecated method.